docs: add intro page for prebuilt patterns (#7958)
carbon-design-system/ibm-products: packages/ibm-products/.storybook/PrebuiltPatterns/About.mdx

import { Meta } from '@storybook/addon-docs/blocks';
import LinkTo from '@storybook/addon-links/react';

<Meta title="Patterns/Prebuilt patterns/About" />

# About prebuilt patterns

## Table of Contents
- [How prebuilt patterns and patterns differ](#how-prebuilt-patterns-and-patterns-differ)
- [Prebuilt patterns to patterns](#prebuilt-patterns-to-patterns)
- [Prebuilt patterns to other alternatives](#prebuilt-patterns-to-other-alternatives)

> Patterns can be accomplished in multiple ways utilizing a combination of component(s) with additional design considerations.

Reviewer: ```suggestion
Patterns can be accomplished in multiple ways utilizing one or more components with additional design considerations.
```

The prebuilt patterns in this library are based on existing [patterns](https://pages.github.ibm.com/carbon/ibm-products/patterns/overview/patterns/)
documented on the Carbon for IBM Products website, however, they are built like traditional components that can also be imported directly for use.

While these prebuilt patterns are available for use directly in code, they only capture a subset of use cases from the broader patterns they represent
and their flexibility may be limited by their component APIs. For a more flexible approach, consider leveraging the equivalent pattern code.

## How prebuilt patterns and patterns differ

Prebuilt patterns can be useful, containing a number of features built-in. However, prebuilt patterns have stricter APIs and may have limited flexibility.

> Many of our patterns include “recipes,” that include a list of components used to deliver the pattern. These are often provided as example code

Reviewer: ```suggestion
Many of our patterns include “recipes” that list the components used to deliver the pattern. These are often provided as example code
```
